File: remove-duplicates-from-sorted-array-1.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int p1 = 1, p2 = 0;
        
        for ( ; p1 < nums.length; p1++ ) {
            if (nums[p1] != nums[p2])
                nums[++p2] = nums[p1];
        }
        
        return p2 + 1;
    }
}

File: remove-duplicates-from-sorted-array-1.scala
object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
        
        if (nums.length < 1)
            return 0
        
        var size = 0
        for (p <- 1 until nums.length) {
            if (nums(p) != nums(size)) {
                size = size + 1
                nums(size) = nums(p)
            }
        }
        
        size + 1
    }
}
